Remove secret permission from the lesser get, list, watch group

lei-tang opened the pull request:
Please provide a description for what this PR is for: https://github.com/istio/istio/issues/19226

lei-tang committed:
Remove secret permission from the lesser get, list, watch group

istio-testing commented:
@lei-tang: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | a1719750930eac54a1d166f2b28e14aa5c691cfb | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/unit-tests_istio/4662) | `/test unit-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@lei-tang: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | a1719750930eac54a1d166f2b28e14aa5c691cfb | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/unit-tests_istio/4662) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | a1719750930eac54a1d166f2b28e14aa5c691cfb | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/pilot-multicluster-e2e_istio/3786) | `/test pilot-multicluster-e2e_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

lei-tang commented:
/retest

istio-testing commented:
@lei-tang: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | a1719750930eac54a1d166f2b28e14aa5c691cfb | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/lint_istio/4528) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

lei-tang commented:
/test lint_istio

istio-testing commented:
@lei-tang: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | a1719750930eac54a1d166f2b28e14aa5c691cfb | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/lint_istio/4540) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

lei-tang commented:
/test lint_istio

istio-testing commented:
@lei-tang: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | a1719750930eac54a1d166f2b28e14aa5c691cfb | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/lint_istio/4553) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@lei-tang: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | c28cf90e2085e5200e8132427625e9d7f5d30552 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/19332/lint_istio/4554) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

lei-tang committed:
Fix lint error

Fix the lint error when running /test lint_istio:

pilot/pkg/model/push_context.go:1514:2: S1023: redundant `return` statement (gosimple)
	return
	^
common/Makefile.common.mk:42: recipe for target 'lint-go' failed
make: *** [lint-go] Error 123

istio-testing commented:
In response to a cherrypick label: #19332 failed to apply on top of branch "release-1.4":
```Applying: Remove secret permission from the lesser get, list, watch group
Applying: Fix lint error
error: Failed to merge in the changes.
Using index info to reconstruct a base tree...
M	pilot/pkg/model/push_context.go
Falling back to patching base and 3-way merge...
Auto-merging pilot/pkg/model/push_context.go
CONFLICT (content): Merge conflict in pilot/pkg/model/push_context.go
Patch failed at 0002 Fix lint error

```
